engine/src/main/java/org/terasology/physics/shapes/CollisionShapeFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.physics.shapes;
+
+import org.terasology.math.geom.Vector3f;
+
+import java.util.List;
+
+/**
+ * Factory interface to create new collision shapes.
+ */
+public interface CollisionShapeFactory {
+    /**
+     * Creates a new box shape with the given extents.
+     * @param extents The full extents of the box shape.
+     * @return The box shape.
+     */
+    BoxShape getNewBox(Vector3f extents);
+
+    /**
+     * Creates a new convex hull shape with the given vertices.
```

Comment:
Do you mean, creates a new convex hull shape _enveloping_ the given vertices or are the vertices meant to be a valid (if unconnected) convex hull already? That is, can this method cope with a generic point cloud or does it require just the outer points of the cloud? 

I'm guessing it is the first option. If that's the case the following line should also be changed in something like: "The vertices to be wrapped by the convex hull."

Reply:
I too support the first option, but will have to check if the `BulletCollisionShapeFactory` implementation of this method follows this norm. I shall update the docs now, but if possible, I shall address the actual implementation in the separate PR dealing with only `CollisionShapes`.